File: vault-web-tests/helpers/pathUtils.ts
export function joinParentName(parentPath: string, name: string): string {
  if (parentPath === '/') {
    return `/${name}`;
  }
  return `${parentPath}/${name}`;
}

export function splitParentName(path: string): [string, string] {
  if (path === '/') {
    throw new Error('cannot call splitParentName for /');
  }
  const parts = path.split('/');
  const name = parts.pop();
  let parentPath = parts.join('/');
  if (parentPath === '') {
    parentPath = '/';
  }
  return [parentPath, name];
}

File: vault-ios/VaultUITests/Helpers/PathUtils.swift
import Foundation

enum PathError: Error {
    case rootPathNotAllowed
    case invalidPath(String)
}

func joinParentName(parentPath: String, name: String) -> String {
    parentPath == "/" ? "/\(name)" : "\(parentPath)/\(name)"
}

func splitParentName(_ path: String) throws -> (String, String) {
    if path == "/" {
        throw PathError.rootPathNotAllowed
    }

    var parts = path.split(separator: "/", omittingEmptySubsequences: false)
        .map(String.init)

    guard let name = parts.popLast(), !name.isEmpty else {
        throw PathError.invalidPath(path)
    }

    var parentPath = parts.joined(separator: "/")
    if parentPath.isEmpty {
        parentPath = "/"
    }

    return (parentPath, name)
}
